datafusion/physical-expr/src/simplifier/unwrap_cast.rs

```diff
@@ -0,0 +1,621 @@
+
+        // Apply unwrap cast optimization
+        let result = unwrap_cast_in_comparison(binary_expr, &schema).unwrap();
+
+        // Should be transformed
+        assert!(result.transformed);
+
+        // Verify the NULL was cast to the column type
+        let optimized = result.data;
+        let optimized_binary = optimized.as_any().downcast_ref::<BinaryExpr>().unwrap();
+        let right_literal = optimized_binary
+            .right()
+            .as_any()
+            .downcast_ref::<Literal>()
+            .unwrap();
+        assert_eq!(right_literal.value(), &ScalarValue::Int32(None));
+    }
+
+    #[test]
+    fn test_unwrap_cast_with_try_cast() {
+        // Test that TryCast expressions are also unwrapped correctly
+        let schema = Schema::new(vec![Field::new("str_col", DataType::Utf8, true)]);
+
+        // Create: try_cast(str_col as INT64) > INT64(100)
+        let column_expr = col("str_col", &schema).unwrap();
+        let try_cast_expr = Arc::new(TryCastExpr::new(column_expr, DataType::Int64));
+        let literal_expr = lit(ScalarValue::Int64(Some(100)));
+        let binary_expr =
+            Arc::new(BinaryExpr::new(try_cast_expr, Operator::Gt, literal_expr));
+
+        // Apply unwrap cast optimization
+        let result = unwrap_cast_in_comparison(binary_expr, &schema).unwrap();
+
+        // Should NOT be transformed (string to int cast not supported)
+        assert!(!result.transformed);
+    }
+
+    #[test]
+    fn test_unwrap_cast_preserves_non_comparison_operators() {
+        // Test that non-comparison operators in AND/OR expressions are preserved
+        let schema = Schema::new(vec![Field::new("int_col", DataType::Int32, false)]);
+
+        // Create: cast(int_col as INT64) > INT64(10) AND cast(int_col as INT64) < INT64(20)
+        let column_expr = col("int_col", &schema).unwrap();
+
+        let cast1 = Arc::new(CastExpr::new(
+            Arc::clone(&column_expr),
+            DataType::Int64,
+            None,
+        ));
+        let lit1 = lit(ScalarValue::Int64(Some(10)));
+        let compare1 = Arc::new(BinaryExpr::new(cast1, Operator::Gt, lit1));
+
+        let cast2 = Arc::new(CastExpr::new(column_expr, DataType::Int64, None));
+        let lit2 = lit(ScalarValue::Int64(Some(20)));
+        let compare2 = Arc::new(BinaryExpr::new(cast2, Operator::Lt, lit2));
+
+        let and_expr = Arc::new(BinaryExpr::new(compare1, Operator::And, compare2));
+
+        // Apply unwrap cast optimization
+        let result = unwrap_cast_in_comparison(and_expr, &schema).unwrap();
+
+        // Should be transformed
+        assert!(result.transformed);
+
+        // Verify the AND operator is preserved
+        let optimized = result.data;
+        let and_binary = optimized.as_any().downcast_ref::<BinaryExpr>().unwrap();
+        assert_eq!(*and_binary.op(), Operator::And);
+
+        // Both sides should have their casts unwrapped
```

Comment:
It might make the tests easier to write / read if you just made the expected output directly

```rust
let expected = lit(10i32)...;
assert_eq!(expected, optimized)
```

Reply:
thanks, done!